### ISSUE TITLE
Full config support in editor settings

### ISSUE THREAD

**Issue by OliverGuy** [astral-sh/ty#1084]
### Summary

It looks like the  [configuration options](https://docs.astral.sh/ty/reference/configuration/) do not seem fully supported within the [editor settings](https://docs.astral.sh/ty/reference/editor-settings/), namely `ty.rules`.

Here's the relevant snippet of my neovim config:
```lua
vim.lsp.config('ty', {
  settings = {
    ty = {
      diagnosticMode = 'workspace',
      experimental = {
        -- this toggle works:
        rename = true,
      },
      rules = {
        -- this one doesn't:
        ['invalid-parameter-default'] = 'ignore',
      },
    },
  },
})
```
Here's a python snippet to trigger the rule I am trying to ignore:
```python
def foo(bar: int = None):
    pass
```

PS: love the project, keep it up 🙏 

### Version

0.0.1-alpha.19 (e9cb838b3 2025-08-19)

**Comment by MichaReiser** [comment]
Adding support for configuring ty directly from within the client is something we want to add but had to deprioritize in favor of supporting more language features. But we'll come back to it. 

This is related to https://github.com/astral-sh/ty/issues/786
